core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/CompactEntityComponent.scala

```diff
@@ -176,6 +182,24 @@ class CompactEntityQuery(driverComponent: DriverComponent) extends RawSqlQuery {
     query.asUpdate
   }
 
+  /**
+   * Get all entity attribute keys for a workspace.
+   */
+  def listEntityKeys(workspaceId: UUID): ReadAction[Seq[EntityTypeAndAttributeKey]] =
+    sql"""SELECT distinct entity_type, attribute_key
+      FROM ENTITY_KEYS , JSON_TABLE(attribute_keys, '$$[*]' COLUMNS(attribute_key VARCHAR(256) PATH '$$')) t
+      where workspace_id=$workspaceId;""".as[EntityTypeAndAttributeKey]
+
+  /**
+   * Gets the count of entities in a workspace, grouped by entity type.
+   */
+  def countEntitiesGroupedByType(workspaceId: UUID): ReadAction[Seq[EntityTypeAndCount]] =
+    // ENTITY_KEYS should be smaller than ENTITY and already excludes deleted entities
+    sql"""SELECT entity_type, COUNT(*)
+      FROM ENTITY_KEYS
+      WHERE workspace_id = $workspaceId
+      GROUP BY entity_type;""".as[EntityTypeAndCount]
+
```

Comment:
nit: I'm trying to set a standard where we include a note about the execution plan in the doc for each query, at least for the Quicksilver stuff - see the other queries. Could you add some details about the execution plans?